src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/ActivityContext.cs

```diff
@@ -59,6 +59,40 @@ public ActivityContext(ActivityTraceId traceId, ActivitySpanId spanId, ActivityT
         /// </remarks>
         public bool IsRemote { get; }
 
+        /// <summary>
+        /// Parse W3C trace parent and state headers to ActivityContext object.
+        /// </summary>
+        /// <param name="traceParent">W3C trace parent header.</param>
+        /// <param name="traceState">Trace state.</param>
+        /// <param name="context">The ActivityContext object created from the parsing operation.</param>
+        public static bool TryParse(string traceParent, string? traceState, out ActivityContext context)
+        {
+            if (traceParent == null)
+            {
+                throw new ArgumentNullException(nameof(traceParent));
```

Comment:
Is throwing okay from "tryParse" methods? Should we instead simply return false?

Reply:
Yes, all TryParse validate the parameters and throws on nulls.